pkg/streamingpromql/limiting/memory_consumption_test.go

```diff
@@ -117,3 +119,81 @@ func createRejectedMetric() (*prometheus.Registry, prometheus.Counter) {
 
 	return reg, metric
 }
+
+func BenchmarkMemoryConsumptionTracker(b *testing.B) {
+	b.Run("no limits single threaded", func(b *testing.B) {
+		l := NewMemoryConsumptionTracker(0, nil)
+		for i := 0; i < b.N; i++ {
+			_ = l.IncreaseMemoryConsumption(uint64(b.N))
+			l.DecreaseMemoryConsumption(uint64(b.N))
+		}
+
+		require.Equal(b, uint64(0), l.CurrentEstimatedMemoryConsumptionBytes())
+	})
+
+	b.Run("with limits single threaded", func(b *testing.B) {
+		l := NewMemoryConsumptionTracker(1024*1024, nil)
+		for i := 0; i < b.N; i++ {
+			if err := l.IncreaseMemoryConsumption(uint64(b.N)); err == nil {
+				l.DecreaseMemoryConsumption(uint64(b.N))
+			}
+		}
+
+		require.Equal(b, uint64(0), l.CurrentEstimatedMemoryConsumptionBytes())
+	})
+
+	b.Run("no limits multiple threads", func(b *testing.B) {
+		l := NewMemoryConsumptionTracker(0, nil)
+		wg := sync.WaitGroup{}
+		run := atomic.NewBool(true)
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for run.Load() {
+				_ = l.IncreaseMemoryConsumption(1024)
+				l.DecreaseMemoryConsumption(1024)
+			}
+		}()
```

Comment:
If we don't use `b.Loop()` below, we should probably call `b.ResetTimer()` after all this setup work is done. Same goes for the test case below.

Reply:
Added calls to `b.ResetTimer()`